array_api_tests/hypothesis_helpers.py

```diff
@@ -164,16 +164,41 @@ def matrix_shapes(draw, stack_shapes=shapes()):
                              elements=dict(allow_nan=False,
                                            allow_infinity=False))
 
-def mutually_broadcastable_shapes(
-    num_shapes: int, **kw
-) -> SearchStrategy[Tuple[Shape, ...]]:
-    return (
-        xps.mutually_broadcastable_shapes(num_shapes, **kw)
-        .map(lambda BS: BS.input_shapes)
-        .filter(lambda shapes: all(
-            prod(i for i in s if i > 0) < MAX_ARRAY_SIZE for s in shapes
-        ))
-    )
+@composite
+def mutually_broadcastable_shapes(draw, num_shapes, **kwargs):
+    # mutually_broadcastable_shapes() with the default inputs doesn't generate
+    # very interesting examples (see
+    # https://github.com/HypothesisWorks/hypothesis/issues/3170). It's very
+    # difficult to get it to do so by tweaking the max_* parameters, because
+    # making them too big leads to generating too large shapes and filtering
+    # too much. So instead, we trick it into generating more interesting
+    # examples by telling it to create shapes that broadcast against some base
+    # shape.
+    if 'base_shape' not in kwargs:
+        base_shape = draw(shapes())
+        kwargs.setdefault('base_shape', base_shape)
+    kwargs.setdefault('min_side', 0)
```

Comment:
```suggestion
    kwargs.setdefault('min_side', 0)
    if 'base_shape' not in kwargs:
        shape_kw = {}
        for k in ['min_dims', 'max_dims', 'min_side', 'max_side']:
            if k in kwargs.keys():
                shape_kw[k] = kwargs[k]
        base_shape = draw(shapes(**shape_kw))
        kwargs['base_shape'] = base_shape
```

Carry over kwargs when generating base shape, avoiding erroneous use of `xps.mutually_broadcastable_shapes()` below.

Reply:
Ah, it hadn't occurred to me that they both take the same keyword arguments.